sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py

```diff
@@ -188,6 +189,62 @@ def _microsoft_teams_user_raw_id(identifier: MicrosoftTeamsUserIdentifier) -> st
     return '8:orgid:{}'.format(user_id)
 
 
+MicrosoftBotProperties = TypedDict(
+    'MicrosoftBotProperties',
+    bot_id=str,
+    is_global=bool,
+    cloud=Union[CommunicationCloudEnvironment, str]
+)
+
+
+class MicrosoftBotIdentifier(object):
+    """Represents an identifier for a Microsoft bot.
+
+    :ivar str raw_id: Optional raw ID of the identifier.
+    :ivar kind: The type of identifier.
+    :vartype kind: str or CommunicationIdentifierKind
+    :ivar Mapping properties: The properties of the identifier.
+     The keys in this mapping include:
+        - `bot_id`(str): The id of the Microsoft bot.
+        - `is_global` (bool): Set this to true if the bot is global.
+        - `cloud` (str): Cloud environment that this identifier belongs to.
+
+    :param str bot_id: Microsoft bot id.
+    :keyword bool is_global: `True` if the identifier is global. Default value is `False`.
+    :keyword cloud: Cloud environment that the bot belongs to. Default value is `PUBLIC`.
+    :paramtype cloud: str or ~azure.communication.chat.CommunicationCloudEnvironment
+    """
+    kind = CommunicationIdentifierKind.MICROSOFT_BOT
+
+    def __init__(self, bot_id, **kwargs):
+        # type: (str, Any) -> None
+        self.raw_id = kwargs.get('raw_id')
+        self.properties = MicrosoftBotProperties(
+            bot_id=bot_id,
+            is_global=kwargs.get('is_global', False),
+            cloud=kwargs.get('cloud') or CommunicationCloudEnvironment.PUBLIC
+        )
+        if self.raw_id is None:
+            self.raw_id = _microsoft_bot_raw_id(self)
+
+
+def _microsoft_bot_raw_id(identifier: MicrosoftBotIdentifier) -> str:
+    bot_id = identifier.properties['bot_id']
+    cloud = identifier.properties['cloud']
+    if identifier.properties['is_global']:
+        if cloud == CommunicationCloudEnvironment.DOD:
+            return '28:dod-global:{}'.format(bot_id)
```

Comment:
we can use f-string formatting as we support py 3.7 +, same for all formatted strings